pyfar/dsp/filter.py

```diff
@@ -976,3 +976,189 @@ def _coefficients_fractional_octave_bands(
                 order, Wn, btype=btype, output='sos')
         sos[idx, :, :] = sos_coeff
     return sos
+
+
+def reconstructing_fractional_octave_bands(
+        signal, num_fractions=1, frequency_range=(63, 16000),
+        overlap=1, slope=0, n_samples=2**12, sampling_rate=None):
+    """
+    Perfectly reconstructing fractional octave filter bank.
+
+    The filters have a linear phase with a delay of ``n_samples/2`` and are
+    windowed with a Hann window to suppress side lobes of the finite filters.
+    The magnitude response of the filters is designed similar to [#]_ with
+    two exceptions:
+
+    1. The magnitude response is designed using squared sine/cosine ramps to
+       obtain -6 dB at the cut-off frequencies.
+    2. The overlap between the filters is calculated between the center and
+       upper cut-off frequencies and not between the center and lower cut-off
+       frequencies. This enables smaller pass-bands with unity gain, which
+       might be advantageous for applications that apply analysis and
+       resynthesis.
+
+    Parameters
+    ----------
+    signal : Signal, None
+        The Signal to be filtered. Pass None to create the filter without
+        applying it.
+    num_fractions : int, optional
+        Octave fraction, e.g., 3 for third-octave bands. The default is ``1``.
+    frequency_range : tuple, optional
+        frequency range for fractional octave in Hz. The default is
+        ``(63, 16000)``
+    overlap : float
+        Band overlap of the filter slopes between 0 and 1. Smaller values yield
+        wider pass-bands and steeper filter slopes. The default is ``1``.
+    slope : int, optional
+        Number > 0 that defines the width and steepness of the filter slopes.
+        Larger values yield wider pass-bands and steeper filter slopes. The
+        default is ``0``.
+    n_samples : int, optional
+        Length of the filter in samples. Longer filters yield more exact
+        filters. The default is ``2**12``.
+    sampling_rate : int
+        Sampling frequency in Hz. The default is ``None``. Only required if
+        ``signal=None``.
+
+    Returns
+    -------
+    signal : Signal
+        The filtered signal. Only returned if ``sampling_rate = None``.
+    filter : FilterFIR
+        FIR Filter object. Only returned if ``signal = None``.
+    frequencies : np.ndarray
+        Center frequencies of the filters.
+
+    References
+    ----------
+    .. [#] Antoni, J. (2010). Orthogonal-like fractional-octave-band filters.
+           J. Acous. Soc. Am., 127(2), 884–895, doi: 10.1121/1.3273888
+
+    Examples
+    --------
+
+    Filter and re-synthesize impulse signal
+
+    >>> import pyfar as pf
+    >>> import numpy as np
+    >>>
+    >>> x = pf.signals.impulse(2**12)
+    >>> y, f = pf.dsp.filter.reconstructing_fractional_octave_bands(x)
+    >>> y_sum = pf.Signal(np.sum(y.time, 0), y.sampling_rate)
+    >>>
+    >>> # plot
```

Comment:
```suggestion
```
Inline comments shouldn't really be required in such a short snippet. Documentation can be added as text above.

Reply:
Removed the inline comments but left blank lines between blocks for better readability